src/client/renderingcore.cpp

```diff
@@ -0,0 +1,386 @@
+	right = driver->addRenderTargetTexture(image_size, "3d_render_right", irr::video::ECF_A8R8G8B8);
+	hud = driver->addRenderTargetTexture(screensize, "3d_render_hud", irr::video::ECF_A8R8G8B8);
+}
+
+void RenderingCoreSideBySide::clear_textures()
+{
+	driver->removeTexture(left);
+	driver->removeTexture(right);
+	driver->removeTexture(hud);
+}
+
+void RenderingCoreSideBySide::update_screen_size()
+{
+	clear_textures();
+	init_textures();
+}
+
+void RenderingCoreSideBySide::draw()
+{
+	pre_draw();
+	draw_two();
+	driver->setRenderTarget(hud, true, true, video::SColor(0, 0, 0, 0));
+	draw_hud();
+	driver->setRenderTarget(nullptr, false, false, skycolor);
+
+	driver->draw2DImage(left, v2s32(0, 0));
+	driver->draw2DImage(right, v2s32(screensize.X / 2, 0));
+
+	driver->draw2DImage(hud,
+			irr::core::rect<s32>(0, 0, screensize.X / 2, screensize.Y),
+			irr::core::rect<s32>(0, 0, screensize.X, screensize.Y), 0, 0,
+			true);
+
+	driver->draw2DImage(hud,
+			irr::core::rect<s32>(screensize.X / 2, 0, screensize.X, screensize.Y),
+			irr::core::rect<s32>(0, 0, screensize.X, screensize.Y), 0, 0,
+			true);
+}
+
+void RenderingCoreSideBySide::use_eye(bool _right)
+{
+	driver->setRenderTarget(_right ? right : left, true, true, skycolor);
+	RenderingCoreStereo::use_eye(_right);
+}
+
+void RenderingCoreSideBySide::reset_eye()
+{
+	driver->setRenderTarget(nullptr, false, false, skycolor);
+	RenderingCoreStereo::reset_eye();
+}
+
+RenderingCorePageflip::RenderingCorePageflip(irr::IrrlichtDevice *_device) :
+	RenderingCoreStereo(_device)
+{
+	init_textures();
+}
+
+RenderingCorePageflip::~RenderingCorePageflip()
+{
+	clear_textures();
+}
+
+void RenderingCorePageflip::init_textures()
+{
+	hud = driver->addRenderTargetTexture(screensize, "3d_render_hud", irr::video::ECF_A8R8G8B8);
+}
+
+void RenderingCorePageflip::clear_textures()
+{
+	driver->removeTexture(hud);
+}
+
+void RenderingCorePageflip::update_screen_size()
+{
+	clear_textures();
+	init_textures();
+}
+
+void RenderingCorePageflip::draw()
+{
+	pre_draw();
+	driver->setRenderTarget(hud, true, true, video::SColor(0, 0, 0, 0));
+	draw_hud();
+	driver->setRenderTarget(nullptr, false, false, skycolor);
+	draw_two();
+}
+
+void RenderingCorePageflip::use_eye(bool _right)
+{
+	driver->setRenderTarget(
+		_right ? irr::video::ERT_STEREO_RIGHT_BUFFER : irr::video::ERT_STEREO_LEFT_BUFFER,
+		 true, true, skycolor);
+	RenderingCoreStereo::use_eye(_right);
+}
+
+void RenderingCorePageflip::reset_eye()
+{
+	driver->draw2DImage(hud, v2s32(0, 0));
+	driver->setRenderTarget(irr::video::ERT_FRAME_BUFFER, false, false, skycolor);
+	RenderingCoreStereo::reset_eye();
+}
+
+RenderingCoreInterlaced::RenderingCoreInterlaced(irr::IrrlichtDevice *_device) :
+	RenderingCoreStereo(_device)
+{
+	mat.MaterialType = (video::E_MATERIAL_TYPE)0;
+	init_textures();
+}
+
+RenderingCoreInterlaced::~RenderingCoreInterlaced()
+{
+	clear_textures();
+}
+
+void RenderingCoreInterlaced::init_material()
+{
+	if (mat.MaterialType)
+		return;
```

Comment:
This is dirty but necessary currently, as smth is NULL at core construction. Really, it should be delayed to game start, when everything is present; that way, settings change wont’t require full restart as well.